src/components/FullDetailView/FullDetailPage.js

```diff
@@ -165,6 +171,45 @@ export class FullDetailPage extends Component {
     })
   }
 
+  handleRevert(value) {
+    const { uiCurateResetDefinitionPreview } = this.props
+    const { changes } = this.state
+    if (isEmpty(changes)) return
+    if (value) {
+      const revertedChanges = omitBy(changes, (_, index) => index.startsWith(value))
+      this.setState({ changes: revertedChanges, sequence: this.state.sequence + 1 }, () => this.previewDefinition())
+      return
+    }
+    const key = `open${Date.now()}`
+    const NotificationButtons = (
```

Comment:
feels like this should be some reuseable notification/confirmer or at least a separate function (renderConfirm or some such)

Reply:
Makes sense, we could definitely improve this.
I will commit a solution into this PR